File: Case Maroon_Backend/src/models/Buildings/Building.js


export class Building {

    static reconScope = 3;
    static reconLevel = 5;
    
    constructor(buildingType, gridPosition) {
         
        this.buildingType = buildingType; // building type
        this.playerId = null; // Player ID who owns this building
        this.gridPosition = gridPosition; // Vector2Int
    }

    toJSON() {
        return {
            id: this.id,
            gridPost: this.position,
            ownerId: this.ownerId,
        };
    } 
}


export const BuildingType = Object.freeze({
    Headquarters: "Headquarters",
    SupplyDepot: "SupplyDepot",
});

File: Case Maroon_Unity/Assets/Scripts/CaseMaroon/Buildings/Building.cs
using System;
using UnityEngine;
using static CaseMaroon.Miscellaneous.GlobalData;

namespace CaseMaroon.Units
{
    [Serializable]
    public class Building
    {
        public Vector2Int gridPosition;
        public BuildingType buildingType;

        public Building(Vector2Int gridPosition, BuildingType buildingType)
        {
            this.gridPosition = gridPosition;
            this.buildingType = buildingType;
        }
    }
}
